aws/resource_aws_elasticache_replication_group.go

```diff
@@ -911,3 +706,218 @@ func validateAwsElastiCacheReplicationGroupEngine(v interface{}, k string) (ws [
+				}
+			}
+
+			// Set new primary
+			err = resourceAwsElasticacheReplicationGroupSetPrimaryClusterID(conn, replicationGroupID, newPrimaryClusterID, timeout)
+			if err != nil {
+				return fmt.Errorf("error changing Elasticache Replication Group (%s) primary cluster: %w", replicationGroupID, err)
+			}
+
+			// Finally retry deleting the cache cluster
+			var finalSnapshotID = d.Get("final_snapshot_identifier").(string)
+			err = deleteElasticacheCacheCluster(conn, cacheClusterID, finalSnapshotID)
+			if err != nil {
+				return fmt.Errorf("error deleting ElastiCache Cache Cluster (%s) (removing replica after setting new primary): %w", cacheClusterID, err)
+			}
+		}
+	}
+
+	// Wait for all Cache Cluster deletions
+	for _, cacheClusterID := range removeClusterIDs {
+		_, err := waiter.CacheClusterDeleted(conn, cacheClusterID, timeout)
+		if err != nil {
+			return fmt.Errorf("error waiting for ElastiCache Cache Cluster (%s) to be deleted (removing replica): %w", cacheClusterID, err)
+		}
+	}
+
+	// Re-enable automatic failover if we needed to temporarily disable it
+	if reEnableAutomaticFailover {
+		err := resourceAwsElasticacheReplicationGroupEnableAutomaticFailover(conn, replicationGroupID, timeout)
+		if err != nil {
+			return fmt.Errorf("error re-enabling Elasticache Replication Group (%s) automatic failover: %w", replicationGroupID, err)
+		}
+	}
+
+	return nil
+}
+
+func resourceAwsElasticacheReplicationGroupDisableAutomaticFailover(conn *elasticache.ElastiCache, replicationGroupID string, timeout time.Duration) error {
+	return resourceAwsElasticacheReplicationGroupModify(conn, timeout, &elasticache.ModifyReplicationGroupInput{
+		ReplicationGroupId:       aws.String(replicationGroupID),
+		ApplyImmediately:         aws.Bool(true),
+		AutomaticFailoverEnabled: aws.Bool(false),
+	})
+}
+
+func resourceAwsElasticacheReplicationGroupEnableAutomaticFailover(conn *elasticache.ElastiCache, replicationGroupID string, timeout time.Duration) error {
+	return resourceAwsElasticacheReplicationGroupModify(conn, timeout, &elasticache.ModifyReplicationGroupInput{
+		ReplicationGroupId:       aws.String(replicationGroupID),
+		ApplyImmediately:         aws.Bool(true),
+		AutomaticFailoverEnabled: aws.Bool(true),
+	})
+}
+
+func resourceAwsElasticacheReplicationGroupSetPrimaryClusterID(conn *elasticache.ElastiCache, replicationGroupID, primaryClusterID string, timeout time.Duration) error {
+	return resourceAwsElasticacheReplicationGroupModify(conn, timeout, &elasticache.ModifyReplicationGroupInput{
+		ReplicationGroupId: aws.String(replicationGroupID),
+		ApplyImmediately:   aws.Bool(true),
+		PrimaryClusterId:   aws.String(primaryClusterID),
+	})
+}
+
+func resourceAwsElasticacheReplicationGroupModify(conn *elasticache.ElastiCache, timeout time.Duration, input *elasticache.ModifyReplicationGroupInput) error {
+	_, err := conn.ModifyReplicationGroup(input)
+	if err != nil {
+		return err
```

Comment:
Nit: Since there is two operations in this function, may be best to return error message context:

```suggestion
		return fmt.Errorf("error modifying ElastiCache Replication Group (%s): %w", aws.StringValue(input.ReplicationGroupId))
```

Reply:
In this case, I was thinking of letting the error bubble up to the top level (at line 609). Though I can see the benefit to clarifying modify vs wait. If we report the error at multiple levels, we get something like:
> error updating ElastiCache Replication Group (abc123): error modifying ElastiCache Replication Group (abc123): Some API error